### PR TITLE
deployment-template: Adds a new --template-format flag to command

### PR DESCRIPTION
**Pull request by karencfv**
## Description
This patch adds a new `--template-format` flag to the `ecctl platform deployment-template {list | show | pull}` commands.

This flag will allow users to specify which inofrmation is populated in the response.
If "cluster" is specified, cluster_template is populated in the response. If deployment is specified, deployment_template is populated in the response.

Requires https://github.com/elastic/cloud-sdk-go/pull/162 to be merged

## Related Issues
Related https://github.com/elastic/cloud-sdk-go/issues/159

## How Has This Been Tested?
manually by running the commands

## Types of Changes

- [x] New feature (non-breaking change which adds functionality)

## Readiness Checklist

- [x] My code follows the code style of this project
- [x] My change requires a change to the documentation
- [x] I have updated the documentation accordingly
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed

### CODE REVIEW

File: cmd/platform/deployment-template/show_test.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+			name: "unspecified region returns error",
+			args: testutils.Args{
+				Cmd: showCmd,
+				Args: []string{
+					"show", "default", "--format=hi",
+				},
+			},
+			want: testutils.Assertion{
+				Err: multierror.NewPrefixed("invalid deployment template get params",
+					errors.New("region not specified and is required for this operation"),
```
Comment (karencfv) [inline cmd/platform/deployment-template/show_test.go:56]: @marclop I'm not quite sure why I'm getting an error here 🤔 Looking at the code for the `testutils` package, the region should be filled by default with `ece-region` if I don't specify the region. But when I try adding `"--region=ece-region"` in the args, it returns `unknown flag: --region`. 

I tried looking around but got stumped, got any ideas?

File: cmd/platform/deployment-template/list.go [hunk 2]
```diff
@@ -55,5 +57,6 @@ func init() {
 	Command.AddCommand(listCmd)
 	listCmd.Flags().BoolP(showInstanceConfigurations, "", false, "Shows instance configurations - only visible when using the JSON output")
 	listCmd.Flags().String(stackVersion, "", "If present, it will cause the returned deployment templates to be adapted to return only the elements allowed in that version.")
-	listCmd.Flags().String(filter, "", "Optional key/value pair in the form of key:value that will act as a filter and exclude any templates that do not have a matching metadata item associated")
+	listCmd.Flags().String(filter, "", "Optional key/value pair in the form of key:value that will act as a filter and exclude any templates that do not have a matching metadata item associated.")
+	listCmd.Flags().String(format, "deployment", "If deployment is specified deployment_template is populated in the response, If cluster is specified cluster_template is populated in the response. (Defaults to deployment)")
```
Comment (nrichers) [inline cmd/platform/deployment-template/list.go:61]:
```suggestion
	listCmd.Flags().String(format, "deployment", "If a deployment is specified, the deployment_template is populated in the response. If cluster is specified, the cluster_template is populated in the response. (Defaults to deployment.)")
```

(We should not omit indefinite and definite articles, as they can help a global audience. The text should also be easier to parse with some added punctuation.)

File: cmd/platform/deployment-template/pull.go [hunk 3]
```diff
@@ -42,4 +45,5 @@ func init() {
 	Command.AddCommand(pullCmd)
 	pullCmd.Flags().StringP("path", "p", "", "Local path where to store deployment templates")
 	pullCmd.MarkFlagRequired("path")
+	pullCmd.Flags().String(format, "deployment", "If deployment is specified deployment_template is populated in the response, If cluster is specified cluster_template is populated in the response. (Defaults to deployment)")
```
Comment (nrichers) [inline cmd/platform/deployment-template/pull.go:48]:
```suggestion
	pullCmd.Flags().String(format, "deployment", "If a deployment is specified, the deployment_template is populated in the response. If a cluster is specified, the cluster_template is populated in the response. (Defaults to deployment.)")
```

(Similar comment as higher up.)

File: cmd/platform/deployment-template/show.go [hunk 4]
```diff
@@ -50,4 +53,5 @@ var showCmd = &cobra.Command{
 func init() {
 	Command.AddCommand(showCmd)
 	showCmd.Flags().BoolP(showInstanceConfigurations, "", false, "Shows instance configurations")
+	showCmd.Flags().String(format, "deployment", "If deployment is specified deployment_template is populated in the response, If cluster is specified cluster_template is populated in the response. (Defaults to deployment)")
```
Comment (nrichers) [inline cmd/platform/deployment-template/show.go:56]:
```suggestion
	showCmd.Flags().String(format, "deployment", "If a deployment is specified, the deployment_template is populated in the response. If a cluster is specified, the cluster_template is populated in the response. (Defaults to deployment.)")
```

(Same comment as higher up.)

File: cmd/platform/deployment-template/show_test.go (hunk 1 above)
Comment (marclop) [reply cmd/platform/deployment-template/show_test.go]: I'll dig in see if I can find it

Comment (marclop) [reply cmd/platform/deployment-template/show_test.go]: I've found the bug and raised https://github.com/elastic/ecctl/pull/336 which should fix this bug.

Comment (karencfv) [reply cmd/platform/deployment-template/show_test.go]: 🙏

**Comment by karencfv** [comment]
Thanks @nrichers  for the comments, I've updated the text :) And thanks @marclop for fixing that bug!

File: cmd/platform/deployment-template/list_test.go [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+]
+`
+
+	tests := []struct {
+		name string
+		args testutils.Args
+		want testutils.Assertion
+	}{
+		{
+			name: "succeeds showing deployment template",
```
Comment (marclop) [inline cmd/platform/deployment-template/list_test.go:71]:
```suggestion
			name: "succeeds listing deployment template",
```

**Pull request merged**